cmd/copyUtil.go

```diff
@@ -55,26 +54,21 @@ func (copyHandlerUtil) numOfWildcardInURL(url url.URL) int {
 	return strings.Count(url.String(), wildCard)
 }
 
-// isIPEndpointStyle checkes if URL's host is IP, in this case the storage account endpoint will be composed as:
-// http(s)://IP(:port)/storageaccount/share(||container||etc)/...
-// TODO: Remove this, it can be replaced by SDK's native support for IP endpoint style.
-func (util copyHandlerUtil) isIPEndpointStyle(url url.URL) bool {
-	return net.ParseIP(url.Host) != nil
-}
-
 // checks if a given url points to a container, as opposed to a blob or prefix match
 func (util copyHandlerUtil) urlIsContainerOrShare(url *url.URL) bool {
-	// When it's IP endpoint style, if the path contains more than two "/", then it means it points to a blob, and not a container.
-	// When it's not IP endpoint style, if the path contains more than one "/", then it means it points to a blob, and not a container.
-	numOfSlashes := strings.Count(url.Path[1:], "/")
-	isIPEndpointStyle := util.isIPEndpointStyle(*url)
-
-	if (!isIPEndpointStyle && numOfSlashes == 0) || (isIPEndpointStyle && numOfSlashes == 1) {
-		return true
-	} else if ((!isIPEndpointStyle && numOfSlashes == 1) || (isIPEndpointStyle && numOfSlashes == 2)) && strings.HasSuffix(url.Path, "/") { // this checks if container_name/ was given
-		return true
+	if azblob.NewBlobURLParts(*url).IPEndpointStyleInfo.AccountName == "" {
+		//Typical endpoint style
+		//If there's no slashes after the first, it's a container.
+		//If there's a slash on the end, it's a virtual directory/container.
+		//Otherwise, it's just a blob.
+		return strings.HasSuffix(url.Path, "/") || strings.Count(url.Path[1:], "/") == 0
+	} else {
+		//URL endpoint style: https://IP:port/accountname/container
+		//If there's 2 or less slashes after the first, it's a container.
```

Comment:
I'm not getting why the constant here is 2 (rather than 1).  Can you add tests to copyUtil_tests.go that construct an actual IP style endpoint, and exercise this branch of the code.  That will (a) give test coverage to this branch and (b) prove whether it should be 1 or 2.

Reply:
So, IP endpoints don't contain the account name in their host. Instead, they contain the account name in their path.

As such, the path takes 2 slashes or less after the first to reach the container.

`/adeledevtest1/proxysync/` for instance, would be the path to my `proxysync` container on an IP endpoint. In comparison, the URL endpoint would be `adeledevtest1.blob.core.windows.net/proxysync/`.

Reply:
If the constant were just 1, then `/adeledevtest1/proxysync` would be considered a blob by AzCopy, and thus we would not provide the service a file name.

Reply:
So it should have one more slash than a URL endpoint, right?  But the constant is not one greater than what is used in URL endpoints, it is currently _two_ greater.  That doesn't smell right.

Reply:
Good point.